presto-docs/src/main/sphinx/presto_cpp/properties.rst

```diff
@@ -0,0 +1,140 @@
+===============================
+Presto C++ Properties Reference
+===============================
+
+This section describes Presto C++ configuration properties.
+
+The following is not a complete list of all configuration and
+session properties, and does not include any connector-specific
+catalog configuration properties. For information on catalog 
+configuration properties, see :doc:`Connectors </connector/>`.
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Coordinator Properties
+----------------------
+
+Set the following properties in the Presto coordinator to enable its use of Presto C++ workers. 
+
+.. code-block:: none
+
+    native-execution-enabled=true
+    optimizer.optimize-hash-generation=false
+    regex-library=RE2J
+    use-alternative-function-signatures=true
+    experimental.table-writer-merge-operator-enabled=false
+
+``experimental.table-writer-merge-operator-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``true``
+
+  Merge TableWriter output before sending to TableFinishOperator.
+
+``native-execution-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``boolean``
+* **Default value:** ``false``
+
+  This property is required when running Presto C++ workers because of 
+  underlying differences in behavior from Java workers.
+
+``optimizer.optimize-hash-generation``
```

Comment:
Session property for this is `optimize_hash_generation` - https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java#L390

Reply:
I don't think we should encourage/document sessions properties for these. Users are better off setting them in the properties file as they are fixed. Advanced users will know the session option :).

Reply:
@agrawalreetika, @majetideepak, because these properties are fixed I understand @majetideepak's recommendation.

However: if these are ever likely to be capable of being set to something else, I'd like to consider adding the session properties to the doc now to reduce the work for future writers. 

Let me know what you think, please.

Reply:
If available, we usually document the session property along with the config properties.
ref - https://prestodb.io/docs/current/admin/properties.html#experimental-spill-enabled 

And I think having a Note along with the property would be better if it has to be something specific for Prestissimo.  WDYT @majetideepak @steveburnett ?

Reply:
Good point @agrawalreetika! We have an established pattern of documenting session properties as part of the configuration property in Presto, it makes sense that the Presto C++ docs should follow the same pattern to help readers. 

Also, while as @majetideepak says "Advanced users will know the session option", I would argue that we shouldn't be writing the doc for only advanced readers. Let's help readers become advanced, by making them aware of the options.

Here, we have a special context of config properties that must (at least currently) be set to a fixed value. To address this concern, we could document the session property by adding something like the following sentence to the definition part of ``optimizer.optimize_hash_generation``:

"Optionally, use the session property ``optimize_hash_generation = false``." 

(I wouldn't suggest following the exact phrasing in https://prestodb.io/docs/current/admin/properties.html#experimental-spill-enabled of "This config property can be overridden by the [x] property." because at this time, the value is fixed, and we don't want them to override the correctly set config property. But if, for some reason, they don't want to or can't set the config property the way we want them to, they can use the session property.)


tl;dr: I think we should document the session properties, and I think the way I'm suggesting addresses the raised issues. Let me know what you think, please.